robot-core/src/main/java/org/obolibrary/robot/checks/Violation.java

```diff
@@ -1,29 +1,69 @@
 package org.obolibrary.robot.checks;
 
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
+import java.util.*;
+import org.semanticweb.owlapi.model.OWLEntity;
+import org.semanticweb.owlapi.model.OWLObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 public class Violation {
 
-  public String subject;
-  public Map<String, List<String>> statements;
+  private final Logger logger = LoggerFactory.getLogger(Violation.class);
 
-  public Violation(String subject) {
-    this.subject = subject;
+  public OWLEntity entity;
+  public Map<OWLEntity, List<OWLObject>> statements;
+
+  /**
+   * Create a new Violation object about an OWL entity.
+   *
+   * @param subject OWLEntity that is the subject of the violation
+   */
+  public Violation(OWLEntity subject) {
+    this.entity = subject;
     this.statements = new HashMap<>();
   }
 
-  public void addStatement(String property, String value) {
-    List<String> values = new ArrayList<>();
+  /**
+   * Add a statement to the Violation about the subject.
+   *
+   * @param property OWLEntity property
+   * @param value OWLObject value
+   */
+  public void addStatement(OWLEntity property, OWLObject value) {
+    List<OWLObject> values;
     if (statements.get(property) != null) {
-      values.addAll(statements.get(property));
+      // This property already has one value in the map
+      // Add this value to the existing list
+      values = new ArrayList<>(statements.get(property));
       values.add(value);
     } else {
-      values = Arrays.asList(value);
+      // This property is not yet in the map
+      // Instantiate a new list
+      if (value != null) {
+        values = Collections.singletonList(value);
+      } else {
+        values = Collections.emptyList();
+      }
     }
     statements.put(property, values);
   }
+
+  @Deprecated public String subject;
+
+  /** @param subject String subject */
+  @Deprecated
+  public Violation(String subject) {
```

Comment:
We published these public methods, so I'd very much like them to continue to work somehow.

Reply:
Makes sense. I've added these back in and added support for these types of violations when printing reports out.